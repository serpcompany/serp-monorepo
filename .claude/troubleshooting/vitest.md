https://vitest.dev/guide/common-errors.html