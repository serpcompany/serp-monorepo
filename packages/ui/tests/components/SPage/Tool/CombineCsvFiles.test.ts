import { mockNuxtImport } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'
import SPageToolCombineCsvFiles from '../../../../components/SPage/Tool/CombineCsvFiles.vue'
import ComponentRender from '../../../componentRender'

mockNuxtImport('useSeoMeta', () => () => {})

describe('sPageToolCombineCsvFiles Snapshot', () => {
  it('renders the default snapshot correctly', async () => {
    const html = await ComponentRender(
      'SPageToolCombineCsvFiles',
      {},
      SPageToolCombineCsvFiles,
    )
    expect(html).toMatchSnapshot()
  })
})
