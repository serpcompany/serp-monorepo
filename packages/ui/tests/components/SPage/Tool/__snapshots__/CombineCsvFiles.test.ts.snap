// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sPageToolCombineCsvFiles Snapshot > renders the default snapshot correctly 1`] = `
<div class="container">
  <div class="
    md:px-8
    md:py-20
    px-4
    py-10
    sm:px-6
    sm:py-16
  ">
    <section class="py-20">
      <h1 class="md:text-center">
        Combine CSV Files
      </h1>
      <p class="
        md:text-center
        sm:text-md
        text-base
        text-left
      ">
        Merge multiple spreadsheet or excel files like .csv, .xls and .xlxs.
      </p>
    </section>
    <!-- box -->
    <div class="
      flex
      justify-center
    ">
      <div class="
        inline-flex
        items-center
        mb-10
        relative
      ">
        <input
          accept=".csv,.xls,.xlsx"
          autocomplete="off"
          class="
            bg-default
            border-0
            disabled:cursor-not-allowed
            disabled:opacity-75
            file:font-medium
            file:me-1.5
            file:outline-none
            file:text-muted
            focus-visible:ring-2
            focus-visible:ring-inset
            focus-visible:ring-primary
            focus:outline-none
            gap-1.5
            placeholder:text-dimmed
            px-2.5
            py-1.5
            ring
            ring-accented
            ring-inset
            rounded-md
            text-highlighted
            text-sm
            transition-colors
            w-full
          "
          multiple=""
          type="file"
        />
        <!-- v-if -->
        <!-- v-if -->
      </div>
    </div>
    <!-- button -->
    <button
      class="
        aria-disabled:bg-primary
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-primary
        disabled:bg-primary
        disabled:cursor-not-allowed
        disabled:opacity-75
        flex
        focus-visible:outline-2
        focus-visible:outline-offset-2
        focus-visible:outline-primary
        font-medium
        gap-1.5
        hover:bg-primary/75
        items-center
        mx-auto
        px-2.5
        py-1.5
        rounded-md
        text-inverted
        text-sm
        transition-colors
      "
      type="button"
    >
      <!-- v-if -->
      Submit
      <!-- v-if -->
    </button>
  </div>
</div>
`;
